### PR TITLE
(PCP-255) Better logging

### PR DESCRIPTION
**Pull request by parisiale**

### CODE REVIEW

File: lib/src/module.cc [hunk 1]
```diff
@@ -24,32 +24,31 @@ ActionOutcome Module::executeAction(const ActionRequest& request) {
...
-            if (!outcome.std_err.empty()) {
-                err_msg += " - stderr: " + outcome.std_err;
-            }
-            throw Module::ProcessingError { err_msg + outcome.std_err };
+            throw Module::ProcessingError {
+                (boost::format("The task executed for the %1% returned %2% "
+                               "results on stdout%3% - stderr:%4%")
+                    % request.prettyLabel()
+                    % (outcome.std_out.empty() ? "no" : "invalid")
+                    % (outcome.std_out.empty() ? "" : std::string(": ") + outcome.std_out)
```
Comment (mruzicka) [inline lib/src/module.cc]: is the ```std::string``` necessary here?

Comment (parisiale) [inline lib/src/module.cc]: No. As long as one argument of operator+ is a string.
Updated.

File: lib/src/external_module.cc [hunk 2]
```diff
@@ -273,24 +273,21 @@ ActionOutcome ExternalModule::processRequestOutcome(const ActionRequest& request
...
 
     if (exit_code != EXIT_SUCCESS) {
-        if (!err_txt.empty()) {
-            LOG_ERROR("'%1% %2%' failure, returned %3%; error: %4%",
-                      module_name, action_name, exit_code, err_txt);
-        } else {
-            LOG_ERROR("'%1% %2%' failure, returned %3%",
-                      module_name, action_name, exit_code);
-        }
+        LOG_TRACE("Execution failure (exit code %1%) for the %2%%3%",
```
Comment (mruzicka) [inline lib/src/external_module.cc]: Should there be a separator (a colon?) between ```%2%``` and ```%3%```?

Comment (mruzicka) [inline lib/src/external_module.cc]: Oh, I see, its part of the parameter: ```"; stderr:\n" + err_txt```

File: lib/src/external_module.cc [hunk 3]
```diff
@@ -299,14 +296,14 @@ ActionOutcome ExternalModule::processRequestOutcome(const ActionRequest& request
...
-        LOG_ERROR("'%1% %2%' output is not valid JSON: %3%",
-                  module_name, action_name, e.what());
-        std::string err_msg { "'" + module_name + " " + action_name + "' "
-                              "returned invalid JSON"  };
-        if (!err_txt.empty()) {
-            err_msg += " - stderr: " + err_txt;
-        }
-        throw Module::ProcessingError { err_msg };
+        LOG_DEBUG("Obtained invalid JSON on stdout for the %1%: %2%",
+                  request.prettyLabel(), e.what());
```
Comment (mruzicka) [inline lib/src/external_module.cc]: Should we log the malformed stdout too?
We log only the ```lth_jc::data_parse_error``` exception's message, and then we include the stderr in the ```Module::ProcessingError``` exception we throw. But we don't log the offending stdout anywhere.

File: lib/src/external_module.cc [hunk 4]
```diff
@@ -362,12 +355,13 @@ ActionOutcome ExternalModule::callNonBlockingAction(const ActionRequest& request
...
-                  module_name, action_name, request.transactionId(),
-                  (exec.output.empty() ? "" : std::string("; output: ") + exec.output),
-                  (exec.output.empty() ? "" : std::string("; error: ") + exec.error));
-        throw Module::ProcessingError { "failed to write output to files" };
+        // This is unexpected. The outcome of the task will not be
+        // available for future transaction status requests
+        LOG_WARNING("The task process failed to write output on file for the %1%%2%%3%",
+                    request.prettyLabel(),
+                    (exec.output.empty() ? "" : "; stdout: " + exec.output),
+                    (exec.error.empty() ? "" : "; stderr: " + exec.error));
```
Comment (mruzicka) [inline lib/src/external_module.cc]: I'd prefer always including the ```stdout: ``` & ```stderr: ``` even if they are empty - in which case I'd just say ```(empty)``` similarly as above.

File: lib/src/external_module.cc (hunk 3 above)
Comment (parisiale) [inline lib/src/external_module.cc]: :+1:

File: lib/src/external_module.cc (hunk 4 above)
Comment (parisiale) [inline lib/src/external_module.cc]: :+1:

**Comment by mruzicka** [comment]
LGTM :+1:

**Pull request merged**